src/constbuffer_array.c

```diff
@@ -108,6 +108,114 @@ CONSTBUFFER_ARRAY_HANDLE constbuffer_array_create_empty(void)
+            {
+                /*Codes_SRS_CONSTBUFFER_ARRAY_42_003: [ constbuffer_array_create_from_array_array shall allocate memory to hold all of the CONSTBUFFER_HANDLES from buffer_arrays. ]*/
+                result = REFCOUNT_TYPE_CREATE_WITH_EXTRA_SIZE(CONSTBUFFER_ARRAY_HANDLE_DATA, (total_buffer_count) * sizeof(CONSTBUFFER_HANDLE));
+                if (result == NULL)
+                {
+                    /*Codes_SRS_CONSTBUFFER_ARRAY_42_008: [ If there are any failures then constbuffer_array_create_from_array_array shall fail and return NULL. ]*/
+                    LogError("failure in malloc");
+                    /*return as is*/
+                }
+                else
+                {
+                    result->nBuffers = total_buffer_count;
+
+                    uint32_t dest_idx;
+                    uint32_t array_idx;
+                    uint32_t source_idx;
+                    for (dest_idx = 0, array_idx = 0; array_idx < buffer_array_count; ++array_idx)
+                    {
+                        for (source_idx = 0; source_idx < buffer_arrays[array_idx]->nBuffers; ++source_idx, ++dest_idx)
+                        {
+                            /*Codes_SRS_CONSTBUFFER_ARRAY_42_004: [ constbuffer_array_create_from_array_array shall copy all of the CONSTBUFFER_HANDLES from each const buffer array in buffer_arrays to the newly constructed array by calling CONSTBUFFER_Clone. ]*/
+                            result->buffers[dest_idx] = CONSTBUFFER_Clone(buffer_arrays[array_idx]->buffers[source_idx]);
+                            if (result->buffers[dest_idx] == NULL)
+                            {
+                                /*Codes_SRS_CONSTBUFFER_ARRAY_42_008: [ If there are any failures then constbuffer_array_create_from_array_array shall fail and return NULL. ]*/
+                                LogError("failure in CONSTBUFFER_Clone for buffer array %" PRIu32 ", buffer %" PRIu32, array_idx, source_idx);
+                                break;
+                            }
+                        }
+
+                        if (source_idx != buffer_arrays[array_idx]->nBuffers)
+                        {
+                            break;
+                        }
+                    }
+
+                    if (array_idx != buffer_array_count)
+                    {
+                        // Cleanup on partial copy
+                        for (i = 0; i < dest_idx; i++)
+                        {
+                            CONSTBUFFER_Destroy(result->buffers[i]);
+                        }
+                    }
+                    else
+                    {
+                        /*Codes_SRS_CONSTBUFFER_ARRAY_42_007: [ constbuffer_array_create_from_array_array shall succeed and return a non-NULL value. ]*/
+                        goto allOk;
+                    }
+
+                    REFCOUNT_TYPE_DESTROY(CONSTBUFFER_ARRAY_HANDLE_DATA, result);
+                    result = NULL;
```

Comment:
>result = NULL; [](start = 20, length = 14)

I'd put result = NULL before the allOk and remove it from the other failure paths.

Reply:
Consider

---
In reply to: [245158902](https://github.com/Azure/azure-c-shared-utility/pull/248#discussion_r245158902) [](ancestors = 245158902)

Reply:
sure

---
In reply to: [245158926](https://github.com/Azure/azure-c-shared-utility/pull/248#discussion_r245158926) [](ancestors = 245158926,245158902)